### DISCUSSION TITLE
Modal improvements

### DISCUSSION THREAD

**Discussion by JamesBock** [open-austin/BASTA-tfwa#197] [Ideas]
https://github.com/open-austin/BASTA-tfwa/blob/ebacad5055f75cb539dd8a4bff1139bfac4b4f64/tenant-file/portal-app/src/component/image-table-collapse.tsx#L51

This is the general functionality but you should split the modal into its own component so we can reuse it with images in other pages.

https://github.com/open-austin/BASTA-tfwa/blob/ebacad5055f75cb539dd8a4bff1139bfac4b4f64/tenant-file/portal-app/src/component/image-table-collapse.tsx#L63

Use the 'name' field instead of 'thumbnailName' to get the full size image. You will have to use some css styling to make the image an appropriate size. 

https://github.com/open-austin/BASTA-tfwa/blob/ebacad5055f75cb539dd8a4bff1139bfac4b4f64/tenant-file/portal-app/src/component/image-table-collapse.tsx#L85

Try using this icon as a close indicator the modal in the header instead of in the footer->`<i className="fa fa-window-close"></i> `
Ideally, reactstrap should have se where is the user clicks outside of the modal it will close.

**Comment by minpat** [comment]
Hey @JamesBock - thanks for the feedback.  I agree it would be good to put the modal in its own component.  I have a first draft of this (in branch pm/194) - when you get a chance, could you please check the image-table-collapse.tsx?  I am having some trouble with the "onClick" event functionality.  When I have something like a console.log statement, it prints out when I click the image.  But if I try to call ShowModal, or even put any sort of HTML elements in there, it seems to not load anything when I click an image.  I am wondering if this is related to the fact that there are some console errors about the type hierarchy (please see screenshot below) - not sure if these were there before.  If you would prefer that I show you on my computer I am happy to do so.

![image](https://user-images.githubusercontent.com/21338961/137611229-abcd56ce-a75f-4fb9-b53c-6d0218a59281.png)

**Reply by JamesBock** [reply]
I believe those console errors where there previously. 

The issue here is that you don't actually have the modal in the DOM anywhere.  
https://github.com/open-austin/BASTA-tfwa/blob/4625606b48627a3904c4ad19e99cea321f8ea9c2/tenant-file/portal-app/src/component/image-table-collapse.tsx#L52

The function is declared but never called. Additionally, you will need to call it in the DOM not in an onClick event as that won't be inserted into the DOM either.
 
![image](https://user-images.githubusercontent.com/37028244/137633388-54a5a031-ff66-4355-9a16-563555198450.png)

Above shows the ShowModal function being called only if the showImageModal bool is true (I added that piece of state back into this component) and the bool's state is controlled by the onClick event on the cell with the image.

I may have led you astray when I said to make the component reusable. This component is VERY reusable but almost too much so to the point where its just a wrapper around a Modal component. I was thinking of a component that can be reused on other Images specifically.  That way we can reuse the formatting and css styling that is appropriate for Image Modals in our app and they don't need to be restated every time.

I would rename the tenantFileCustomModal to imageModal and the css module to imageModal.module.css and focus on making the Modal presentable for the full size image. You will probably want to explicitly set the max-height and max-width at least.

Also, you can simplify this code.
![image](https://user-images.githubusercontent.com/37028244/137634425-84c46d33-7aa8-42ab-959a-7839d4997220.png)

I used that syntax for creating the download link because if you just call onClick on an image it will navigate to it, not download it. You have to use the anchor tag to download. For onClick events that just need to work like buttons, you can just do this. 
![image](https://user-images.githubusercontent.com/37028244/137634531-e6129f92-3e7a-45c9-ac54-43be03910923.png)

**Reply by JamesBock** [reply]
@minpat

**Reply by minpat** [reply]
Got it - thanks @JamesBock for the info - much appreciated.  I will give this a try and will make the component more specific to images.
